rclcpp/topics/minimal_publisher/member_function_with_unique_network_flow_endpoints.cpp

```diff
@@ -31,29 +31,36 @@ class MinimalPublisherWithUniqueNetworkFlowEndpoints : public rclcpp::Node
   MinimalPublisherWithUniqueNetworkFlowEndpoints()
   : Node("minimal_publisher_with_unique_network_flow_endpoints"), count_1_(0), count_2_(0)
   {
-    // Create publisher with unique network flow endpoints
-    // Enable unique network flow endpoints via options
-    auto options_1 = rclcpp::PublisherOptions();
-    options_1.require_unique_network_flow_endpoints =
-      RMW_UNIQUE_NETWORK_FLOW_ENDPOINTS_OPTIONALLY_REQUIRED;
-    publisher_1_ = this->create_publisher<std_msgs::msg::String>("topic_1", 10, options_1);
-    timer_1_ = this->create_wall_timer(
-      500ms, std::bind(&MinimalPublisherWithUniqueNetworkFlowEndpoints::timer_1_callback, this));
+    try {
+      // Create publisher with unique network flow endpoints
+      // Enable unique network flow endpoints via options
+      auto options_1 = rclcpp::PublisherOptions();
+      options_1.require_unique_network_flow_endpoints =
+        RMW_UNIQUE_NETWORK_FLOW_ENDPOINTS_OPTIONALLY_REQUIRED;
+      publisher_1_ = this->create_publisher<std_msgs::msg::String>("topic_1", 10, options_1);
+      timer_1_ = this->create_wall_timer(
+        500ms, std::bind(&MinimalPublisherWithUniqueNetworkFlowEndpoints::timer_1_callback, this));
 
-    // Create publisher without unique network flow endpoints
-    // Unique network flow endpoints are disabled in default options
-    auto options_2 = rclcpp::PublisherOptions();
-    publisher_2_ = this->create_publisher<std_msgs::msg::String>("topic_2", 10);
-    timer_2_ = this->create_wall_timer(
-      1000ms, std::bind(&MinimalPublisherWithUniqueNetworkFlowEndpoints::timer_2_callback, this));
+      // Create publisher without unique network flow endpoints
+      // Unique network flow endpoints are disabled in default options
+      auto options_2 = rclcpp::PublisherOptions();
+      publisher_2_ = this->create_publisher<std_msgs::msg::String>("topic_2", 10);
+      timer_2_ = this->create_wall_timer(
+        1000ms, std::bind(&MinimalPublisherWithUniqueNetworkFlowEndpoints::timer_2_callback, this));
 
-    // Get network flow endpoints
-    auto network_flow_endpoints_1 = publisher_1_->get_network_flow_endpoints();
-    auto network_flow_endpoints_2 = publisher_2_->get_network_flow_endpoints();
+      // Get network flow endpoints
+      auto network_flow_endpoints_1 = publisher_1_->get_network_flow_endpoints();
+      auto network_flow_endpoints_2 = publisher_2_->get_network_flow_endpoints();
 
-    // Print network flow endpoints
-    print_network_flow_endpoints(network_flow_endpoints_1);
-    print_network_flow_endpoints(network_flow_endpoints_2);
+      // Print network flow endpoints
+      print_network_flow_endpoints(network_flow_endpoints_1);
+      print_network_flow_endpoints(network_flow_endpoints_2);
+    } catch (const rclcpp::exceptions::RCLError & e) {
+      RCLCPP_ERROR(
+        this->get_logger(),
+        "Error: %s",
+        e.what());
+    }
```

Comment:
The problem with this is that now the constructor will "succeed", but not actually create all of the resources it should.

My suggestion is to actually not do this `try/except` here, but rather do it in main.  That way if we fail, we can print the error message and quit the program with a non-zero exit code.

Reply:
if that is supposed to mean the publisher, constructor will succeed and all of the resource will be created. but it fails on `get_network_flow_endpoints`. so there would be no problem? btw, i am also good to go with `moving try&catch statement in main`, if that is more comfortable example for user.

Reply:
> if that is supposed to mean the publisher, constructor will succeed and all of the resource will be created. but it fails on `get_network_flow_endpoints`. so there would be no problem?

Oh, I see.  Well, I think the problem here is that since we are catching the generic `RCLError`, many errors (including those creating the publisher or timer) might be caught here.  So with that said, I think there are two ways we can go here:

1. Move the try/catch to main.  That will catch all errors from the constructor.
2. Just put the try/catch around `get_network_flow_endpoints()`, since that is the error we are worried about here.

I'd personally lean towards the second one, but the first one is OK too.

Reply:
> I'd personally lean towards the second one, but the first one is OK too.

okay, i am good to go with either way.